app/cli/internal/policydevel/lint.go

```diff
@@ -0,0 +1,390 @@
+	if os.IsNotExist(err) {
+		return nil, fmt.Errorf("path does not exist: %s", absPath)
+	}
+
+	policy := &Policy{
+		Path:   absPath,
+		Format: format,
+	}
+
+	if fileInfo.IsDir() {
+		// Read all *.yaml and *.rego files in the directory
+		files, err := os.ReadDir(absPath)
+		if err != nil {
+			return nil, fmt.Errorf("reading directory: %w", err)
+		}
+
+		for _, file := range files {
+			if file.IsDir() {
+				continue
+			}
+
+			filePath := filepath.Join(absPath, file.Name())
+			content, err := os.ReadFile(filePath)
+			if err != nil {
+				return nil, fmt.Errorf("reading %s: %w", file.Name(), err)
+			}
+
+			switch strings.ToLower(filepath.Ext(file.Name())) {
+			case ".yaml", ".yml":
+				policy.YAMLFiles = append(policy.YAMLFiles, &File{
+					Path:    filePath,
+					Content: content,
+				})
+			case ".rego":
+				policy.RegoFiles = append(policy.RegoFiles, &File{
+					Path:    filePath,
+					Content: content,
+				})
+			}
+
+			if len(policy.YAMLFiles) == 0 && len(policy.RegoFiles) == 0 {
+				return nil, fmt.Errorf("directory must contain at least one .yaml/.yml or .rego file")
+			}
+		}
+	} else {
+		// Read given file
+		content, err := os.ReadFile(absPath)
+		if err != nil {
+			return nil, fmt.Errorf("reading file: %w", err)
+		}
+
+		switch strings.ToLower(filepath.Ext(absPath)) {
+		case ".yaml", ".yml":
+			policy.YAMLFiles = append(policy.YAMLFiles, &File{
+				Path:    absPath,
+				Content: content,
+			})
+		case ".rego":
+			policy.RegoFiles = append(policy.RegoFiles, &File{
+				Path:    absPath,
+				Content: content,
+			})
+		default:
+			return nil, fmt.Errorf("file must be either .yaml/.yml or .rego")
+		}
+	}
+
+	return policy, nil
+}
+
+func (p *Policy) Validate() []error {
+	var allErrors []error
+
+	// Validate all YAML files (including their embedded policies)
+	for _, yamlFile := range p.YAMLFiles {
+		if yamlErrs := p.validateYAMLFile(yamlFile); len(yamlErrs) > 0 {
+			allErrors = append(allErrors, yamlErrs...)
+		}
+	}
+
+	// Validate standalone rego files
+	for _, regoFile := range p.RegoFiles {
+		if regoErrs := p.validateRegoFile(regoFile); len(regoErrs) > 0 {
+			allErrors = append(allErrors, regoErrs...)
+		}
+	}
+
+	return allErrors
+}
+
+func (p *Policy) validateYAMLFile(file *File) []error {
+	var errors []error
+	var node yaml.Node
+
+	if err := yaml.Unmarshal(file.Content, &node); err != nil {
+		return []error{fmt.Errorf("%s: invalid YAML: %w", file.Path, err)}
+	}
+
+	var data map[string]interface{}
+	if err := node.Decode(&data); err != nil {
+		return []error{fmt.Errorf("%s: invalid YAML structure: %w", file.Path, err)}
+	}
+
+	// Validate YAML structure
+	if missing := checkRequiredFields(data, []string{"apiVersion", "kind", "spec"}); len(missing) > 0 {
+		errors = append(errors, fmt.Errorf("%s: missing required fields: %s", file.Path, strings.Join(missing, ", ")))
+	}
+
+	// Process each embedded policy
+	if spec, ok := data["spec"].(map[string]interface{}); ok {
```

Comment:
I think you should unmarshal against the actual proto spec using protojson.Unmarshal and protovalidaet.Validate.